### PR TITLE
feat(x/ecocredit): sdk.Msg impl for MsgPut

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

Adds the sdk.Msg implementation for MsgPut. 
Adds tests for ValidateBasic.

Ref: #732 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by aaronc — APPROVED** [review]

File: x/ecocredit/basket/msgs.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+func (m MsgPut) Type() string { return sdk.MsgTypeURL(&m) }
+
+func (m MsgPut) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ecocredit.ModuleCdc.MustMarshalJSON(&m))
+}
+
+func (m MsgPut) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Owner); err != nil {
+		return err
```
Comment (robert-zaremba) [inline x/ecocredit/basket/msgs.go:27]: please wrap with ErrInvalidRequest

File: x/ecocredit/basket/msgs.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		return sdkerrors.ErrInvalidRequest.Wrapf("%s is not a valid basket denom", m.BasketDenom)
+	}
+	if len(m.Credits) > 0 {
+		for _, credit := range m.Credits {
+			if err := ecocredit.ValidateDenom(credit.BatchDenom); err != nil {
+				return err
+			}
+			if _, err := math.NewDecFromString(credit.Amount); err != nil {
+				return err
+			}
```
Comment (robert-zaremba) [inline x/ecocredit/basket/msgs.go:39]: How about adding a check that amount is not zero?

Comment (aaronc) [reply x/ecocredit/basket/msgs.go:39]: Must be positive

Comment (technicallyty) [reply x/ecocredit/basket/msgs.go:39]: fixed

File: x/ecocredit/basket/msgs.go (hunk 1 above)
Comment (technicallyty) [reply x/ecocredit/basket/msgs.go]: done

**Comment by ryanchristo** [comment]
We are merging the other pull requests into `release/v2.2.x` - should this do the same?

**Comment by technicallyty** [comment]
> We are merging the other pull requests into `release/v2.2.x` - should this do the same?

yeah i just realized lol ill fix that

File: x/ecocredit/basket/msgs.go (hunk 2 above)
Comment (robert-zaremba) [reply x/ecocredit/basket/msgs.go:39]: not pushed?

**Review by robert-zaremba — APPROVED** [review]
pending `amount > 0` check

File: x/ecocredit/basket/msgs.go (hunk 2 above)
Comment (technicallyty) [reply x/ecocredit/basket/msgs.go:39]: was having issues with switching base, should be all good now! thanks!

**Pull request merged**